tower/src/buffer/service.rs

```diff
@@ -13,18 +13,20 @@ use tower_service::Service;
 ///
 /// See the module documentation for more details.
 #[derive(Debug)]
-pub struct Buffer<T, Request>
+pub struct Buffer<S, Request, E2>
```

Comment:
Can `E2` have a default so that users who don't care about overriding `Box<dyn Error>` don't have to think about it?

Reply:
I'm not sure if we could default it in a meaningful way, the default only comes into play when you're using the constructor itself, not on any functions like `new` that then construct the `Buffer` internally.

Reply:
Yeah, but it also shows up if I want to write code that has a `Buffer` as a field or returns a `Buffer`.

Reply:
I might also want to consider having `new` fill in the default, and have a separate constructor for overriding the error type...but I could be convinced otherwise.

Reply:
> I might also want to consider having new fill in the default, and have a separate constructor for overriding the error type...but I could be convinced otherwise.

I have no desire to convince you otherwise, this sounds like a great idea.

Reply:
And added in the last commit